libraries/Microsoft.Bot.Builder.AI.LUIS/LuisV3Util.cs

```diff
@@ -0,0 +1,193 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Linq;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Bot.Builder.AI.Luis
+{
+    // Utility functions used to extract and transform data from Luis SDK
+    internal static class LuisV3Util
+    {
+        internal const string _metadataKey = "$instance";
+        internal const string _geoV2 = "builtin.geographyV2.";
+        internal static readonly HashSet<string> _dateSubtypes = new HashSet<string> { "date", "daterange", "datetime", "datetimerange", "duration", "set", "time", "timerange" };
+
+        internal static string NormalizedIntent(string intent) => intent.Replace('.', '_').Replace(' ', '_');
+
+        internal static IDictionary<string, IntentScore> GetIntents(JObject luisResult)
+        {
+            IDictionary<string, IntentScore> result = null;
+            var intents = (JObject)luisResult["intents"];
```

Comment:
Can this return an empty dictionary, rather than null, in the default case?

Reply:
Done